activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -230,7 +230,15 @@ def column_for_attribute(name)
     #   person.respond_to(:nothing) # => false
     def respond_to?(name, include_private = false)
       return false unless super
-      name = name.to_s
+
+      case name
```

Comment:
this case statement knows a lot about its ecosystem. in the whole file there is no mention to `to_partial_path` or `to_model`. Could be difficult to read it is just performance. For me this goes too far as optimization